ruby/lib/ci/queue/grind.rb

```diff
@@ -0,0 +1,22 @@
+require 'ci/queue/static'
+
+module CI
+  module Queue
+    class Grind < Static
+      class << self
+        def from_uri(uri, config)
+          new(uri.path, config)
+        end
+      end
+
+      def initialize(path, config)
+        io = path == '-' ? STDIN : ::File.open(path)
+
+        tests_to_run = io.each_line.map(&:strip).reject(&:empty?).to_set
+        test_grinds = Array.new(config.grind_count) { tests_to_run.to_a }.flatten.sort
```

Comment:
So the implementation is very close to `File`.

If you inherited you could simply do:

```ruby
super
@queue *= config.grind_count
```

Reply:
The one thing that makes me a little bit hesitant is that the Static class sets `@total = tests.size` in its initializer, which the File class inherits from. I think that would complicate things, but I might be wrong.

Reply:
RIght. I would at least simplify the building of the repetition. That `Array.new` and the `.to_set` is much more complex that it needs to.